### PR TITLE
added validations

### PR DESCRIPTION
**Pull request by clarasenni**

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -2,4 +2,6 @@ class Item < ApplicationRecord
   belongs_to :user
   has_many :rentals
   has_many :users, through: :rentals
+
+  validates :name, :price, :condition, :brand, :category, presence: true
```
Comment (elizaup) [inline app/models/item.rb:6]: this looks good

File: app/models/item.rb [hunk 2]
```diff
@@ -2,4 +2,6 @@ class Item < ApplicationRecord
   belongs_to :user
   has_many :rentals
   has_many :users, through: :rentals
+
```
Comment (elizaup) [inline app/models/item.rb:5]: can you add this as well:

  ```
validates :condition, inclusion: { in: ["Brand new", "Like new", "Used - Excellent", "Used - Good", "Used - Fair"] }
  validates :condition, inclusion: { in: ["Tops", "Bottoms", "Dressses and Jumpsuits", "Coats and Jackets", "Footwear",
                                          "Sets", "Accessories"] }
```

File: app/models/rental.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 class Rental < ApplicationRecord
   belongs_to :user
   belongs_to :item
+
+  validates :start_date, :end_date, presence: true
```
Comment (elizaup) [inline app/models/rental.rb:5]: looks good

File: app/models/user.rb [hunk 4]
```diff
@@ -5,4 +5,9 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
   has_many :rentals
   has_many :items, through: :rentals
+
+  validates :first_name, :last_name, :username, :adress, :encrypted_password, :email, presence: true
+  validates :first_name, uniqueness: { scope: :last_name }
```
Comment (elizaup) [inline app/models/user.rb:10]: first name can't be unique so I would remove this?

File: app/models/user.rb [hunk 5]
```diff
@@ -5,4 +5,9 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
   has_many :rentals
   has_many :items, through: :rentals
+
+  validates :first_name, :last_name, :username, :adress, :encrypted_password, :email, presence: true
```
Comment (elizaup) [inline app/models/user.rb:9]: address is misspelt, i would suggest taking these names from the schema so you don't make mistakes

`validates :username, :first_name, :last_name, :email, :address, :encrypted_password, presence: true
end`

**Review by elizaup — CHANGES_REQUESTED** [review]
```

```

File: app/models/user.rb (hunk 4 above)
Comment (clarasenni) [reply app/models/user.rb:10]: It means that the combination of that first and last name have to be unique, so noone can have the exact same first and last name, but yes, that might not be necessary.

File: app/models/item.rb (hunk 2 above)
Comment (clarasenni) [reply app/models/item.rb:5]: yes perfect! Only thing is, you assigned both validations to the variable condition, I'm assuming one should be "condition" and the other on"category"?

Comment (elizaup) [reply app/models/item.rb:5]: oh sorry i was duplicating it to avoid errros in syntax and forgot to update my word

File: app/models/user.rb (hunk 4 above)
Comment (elizaup) [reply app/models/user.rb:10]: i see, smart. up to you :)

**Comment by mktraynor** [comment]
Resolved by Eliza's changes

**Pull request closed without merging**